patches/api/0440-SpawnerSearchEvent.patch

```diff
@@ -0,0 +1,131 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TreemanKing <67459602+TreemanKing@users.noreply.github.com>
+Date: Wed, 16 Aug 2023 11:59:08 +1000
+Subject: [PATCH] SpawnerSearchEvent
+
+This patch introduces a new event, SpawnerSearchEvent. This event is designed to be fired before a mob spawner attempts to search for nearby players to activate. It provides the ability to control and customize the activation logic of mob spawners, preventing them from searching and spawning mobs if desired
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/SpawnerSearchEvent.java b/src/main/java/io/papermc/paper/event/entity/SpawnerSearchEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..72b0ae8a84825cf273eba0827fb0551c1bda27e8
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/SpawnerSearchEvent.java
+@@ -0,0 +1,118 @@
++package io.papermc.paper.event.entity;
++
++import com.google.common.base.Preconditions;
++import org.bukkit.Location;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Event;
++import org.bukkit.event.HandlerList;
++import org.jetbrains.annotations.ApiStatus;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Range;
++import org.jetbrains.annotations.Unmodifiable;
++import java.util.Collection;
++import java.util.Collections;
++
++/**
++ * Represents an event that is fired before a mob spawner attempts to search
++ * for nearby players to activate. This event occurs before any of the spawner logic,
++ * allowing cancellation to prevent the spawner from searching and spawning any mobs.
```

Comment:
Am I bind or does it actually not allow cancellation?

Reply:
It does via setResult(DENY)